src/IgnoreList.cxx

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The Music Player Daemon Project
+
+#include "IgnoreList.hxx"
+
+#include <cassert>
+#include <ranges>
+#include <algorithm>
+
+bool IgnoreListEntry::matches_record(const Record& record) const {
+	// The below logic would always return true if the entry is empty.
+	// This condition should never be true, as we don't push empty entries.
+	assert(this->artist || this->album || this->title);
+
+	if (this->artist) {
+		if (record.artist.empty() || record.artist != this->artist) {
+			return false;
+		}
+	}
+
+	if (this->album) {
+		if (record.album.empty() || record.album != this->album) {
+			return false;
+		}
+	}
+
+	if (this->title) {
+		if (record.track.empty() || record.track != this->title) {
+			return false;
+		}
+	}
```

Comment:
That's 3 copies of the same piece of code

Reply:
Hm, would you like me to reduce the boilerplate with a macro? I don't see a better way right now.

Reply:
```
[[gnu::pure]]
static constexpr bool
IsIgnoreMismatch(std::string_view ignore, std::string_view value) noexcept
{
  return !ignore.empty() && (value.empty() || value != ignore);
}
```
Simple, eh?

Reply:
Yeah, true. Will fix this.